src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/DependencyAnalysisState.cs

```diff
@@ -6,14 +6,24 @@
 
 namespace Microsoft.DotNet.UpgradeAssistant.Steps.Packages
 {
-    internal class DependencyAnalysisState : IDependencyAnalysisState
+    public class DependencyAnalysisState : IDependencyAnalysisState
     {
         public DependencyAnalysisState(IProject project, INuGetReferences nugetReferences)
         {
-            FrameworkReferences = new DependencyCollection<Reference>(project.FrameworkReferences, SetRisk);
-            Packages = new DependencyCollection<NuGetReference>(nugetReferences.PackageReferences, SetRisk);
-            References = new DependencyCollection<Reference>(project.References, SetRisk);
+            if (project is null)
+            {
+                throw new ArgumentNullException(nameof(project));
+            }
+
+            if (nugetReferences is null)
+            {
+                throw new ArgumentNullException(nameof(nugetReferences));
+            }
 
+            FrameworkReferences = new DependencyCollection<Reference>(initial: project.FrameworkReferences, setRisk: SetRisk);
+            Packages = new DependencyCollection<NuGetReference>(initial: nugetReferences.PackageReferences, setRisk: SetRisk);
+            References = new DependencyCollection<Reference>(project.References, SetRisk);
+            IsValid = true;
```

Comment:
Should it start as invalid, and then explicitly set as valid?

Reply:
its explicitly set to false when an error occurs, otherwise it is set to true. Incase of null operators, I'm throwing an ArgumentNullException.